src/test/mir-opt/const_allocation3/32bit/rustc.main.ConstProp.after.mir

```diff
@@ -0,0 +1,61 @@
+// MIR for `main`
+// source = MirSource { instance: Item(DefId(0:3 ~ const_allocation3[317d]::main[0])), promoted: None }
+// pass_name = ConstProp
+// disambiguator = after
+
+fn  main() -> () {
+    let mut _0: ();                      // return place in scope 0 at $DIR/const_allocation3.rs:4:11: 4:11
+    let _1: &Packed;                     // in scope 0 at $DIR/const_allocation3.rs:5:5: 5:8
+    let mut _2: &&Packed;                // in scope 0 at $DIR/const_allocation3.rs:5:5: 5:8
+
+    bb0: {
+        StorageLive(_1);                 // bb0[0]: scope 0 at $DIR/const_allocation3.rs:5:5: 5:8
+        StorageLive(_2);                 // bb0[1]: scope 0 at $DIR/const_allocation3.rs:5:5: 5:8
+        _2 = const {alloc4+0: &&Packed}; // bb0[2]: scope 0 at $DIR/const_allocation3.rs:5:5: 5:8
+                                         // ty::Const
+                                         // + ty: &&Packed
+                                         // + val: Value(Scalar(alloc4+0))
+                                         // mir::Constant
+                                         // + span: $DIR/const_allocation3.rs:5:5: 5:8
+                                         // + literal: Const { ty: &&Packed, val: Value(Scalar(alloc4+0)) }
+        _1 = (*_2);                      // bb0[3]: scope 0 at $DIR/const_allocation3.rs:5:5: 5:8
+        StorageDead(_2);                 // bb0[4]: scope 0 at $DIR/const_allocation3.rs:5:8: 5:9
+        StorageDead(_1);                 // bb0[5]: scope 0 at $DIR/const_allocation3.rs:5:8: 5:9
+        _0 = ();                         // bb0[6]: scope 0 at $DIR/const_allocation3.rs:4:11: 6:2
+        return;                          // bb0[7]: scope 0 at $DIR/const_allocation3.rs:6:2: 6:2
+    }
+}
+
+alloc4 (static: FOO, size: 4, align: 4) {
+    ╾alloc14+0╼
+}
+
+alloc14 (size: 168, align: 1) {
+    0x00 │ ab ab ab ab ab ab ab ab ab ab ab ab ab ab ab ab
+    0x10 │ ab ab ab ab ab ab ab ab ab ab ab ab ╾alloc9+0─╼
+    0x20 │ 01 ef cd ab 00 00 00 00 00 00 00 00 00 00 00 00
+    0x30 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+    0x40 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+    0x50 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+    0x60 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+    0x70 │ 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+    0x80 │ 00 00 00 00 00 00 00 00 00 00 ╾alloc11+0╼ 00 00
+    0x90 │ ╾alloc12+99 (4 ptr bytes)╼ 00 00 00 00 00 00 00 00 00 00 00 00
```

Comment:
Here's one silly way to handle this: go through all the relocations first and print them, compute the overflow, then track that overflow per-column.

So that on every row, you can print a number of spaces between the last column of the relocation and the next column.

Reply:
This seems like a pretty hard problem in the presence of enough (badly aligned) relocations.

Reply:
Not really, you have a fixed number of columns so worst case it gets very wide, but it's still (arguably) readable.

There's no fixpoint computation you'd need to do, unless... oh I see.

You'd want to sort relocations by last column (i.e. `(offset + ptr_size - 1) % columns`) and handle them in that order, so that you prefer widening earlier columns than later ones.

Reply:
Would it be ok with you if I worked on this in a separate PR?

Reply:
Oh yeah it's a potentially nice trick but it can wait. Make an issue?

Reply:
#70339